### ISSUE TITLE
horovod MPI_Init_thread required thread level should be MPI_THREAD_SINGLE

### ISSUE THREAD

**Issue by LiweiPeng** [uber/horovod#426]
To support RDMA RoCE, when HOROVOD_MPI_THREADS_DISABLE is defined, MPI_Init_thread required thread level should be MPI_THREAD_SINGLE. 

Here is why: 
- The current horovod implementation won't work with OpenMPI when RDMA RoCE is used. OpenMPI will report error saying 'OpenFabric is disabled etc'.
- OpenMPI uses rdmacm for RDMA RoCE support. However, because rdmacm is not thread-safe, OpenMPI only supports MPI_THREAD_SINGLE if RoCE is used.

Because all the MPI APIs are called in one dedicated thread in horovod, this change won't break horovod design.

**Comment by anpark** [comment]
can you share the details of RDMA + horovod ? such as cluster env, horovod commands and so on.  thanks

**Comment by LiweiPeng** [comment]
@alsrgv I am using openmpi 3.1.1. The logging at btl_openib_connect_rdmacm.c:2021 is not accurate. The actual code logic is: 
- If opal_using_threads() is true, it will have this error.
- opal_using_threads is set at this line: https://github.com/open-mpi/ompi/blob/master/ompi/runtime/ompi_mpi_init.c#L436, like below
```
   ...
    if (*provided != MPI_THREAD_SINGLE) {
        opal_set_using_threads(true);
    }
```

**Comment by LiweiPeng** [comment]
@anpark I am using RDMA RoCE with OpenMPI. The mpi command line looks like below
```
mpirun --machinefile mach.txt --map-by ppr:1:node -x HOROVOD_MPI_THREADS_DISABLE=1  --mca btl openib,self,smcuda --mca btl_openib_cpc_include rdmacm --mca btl_openib_rroce_enable 1 --mca btl_openib_receive_queues P,256,256::S,128,256,192,128:S,2048,1024,1008,64:S,12288,1024,1008,64:S,131072,1024,1008,64 --mca btl_openib_cuda_async_recv false -x NCCL_IB_DISABLE=0 -x NCCL_IB_HCA=mlx5_bond_0 -x NCCL_IB_GID_INDEX=3 -x NCCL_IB_CUDA_SUPPORT=1 test-command
```
I am using cuda 9.2 with GPU direct. The critical part is rdmacm used by OpenMPI for openib.

**Comment by alsrgv** [comment]
@LiweiPeng, good catch.  I have published a PR #447 with a fix, could you try it and confirm it fixes your problem?

**Comment by LiweiPeng** [comment]
This PR #447  works for me. I have been running this fix for about 2 weeks by changing this myself.
